src/control/build/interop_test.go

```diff
@@ -88,6 +88,34 @@ func TestBuild_CheckCompatibility(t *testing.T) {
 			b:      testComponent(t, "", "1.2.0"),
 			expErr: errors.New("incompatible components"),
 		},
+		"2.4 server compatible with 2.2 agent": {
+			a: testComponent(t, "server", "2.4.0"),
+			b: testComponent(t, "agent", "2.2.0"),
+		},
+		"2.4 agent compatible with 2.2 server": {
+			a: testComponent(t, "server", "2.2.0"),
+			b: testComponent(t, "agent", "2.4.0"),
+		},
+		"2.6 server not compatible with 2.2 agent": {
+			a:      testComponent(t, "server", "2.6.0"),
+			b:      testComponent(t, "agent", "2.2.0"),
+			expErr: errors.New("incompatible components"),
+		},
+		"2.6 agent not compatible with 2.2 server": {
+			a:      testComponent(t, "server", "2.2.0"),
+			b:      testComponent(t, "agent", "2.6.0"),
+			expErr: errors.New("incompatible components"),
+		},
+		"3.4 server not compatible with 2.2 agent": {
+			a:      testComponent(t, "server", "3.4.0"),
+			b:      testComponent(t, "agent", "2.2.0"),
+			expErr: errors.New("incompatible components"),
+		},
+		"unversioned agent (assume 2.0.x) compatible with 2.2.x server": {
```

Comment:
wouldn't this fail the "'backwards compatibility prior to 2.0.0" default rule?

Reply:
No, because the custom rules are checked before the general rules, and the custom rule has the `StopOnSuccess` directive set to short-circuit the rule processing logic.